### PR TITLE
Update packaging bits to build scst-dbg package

### PR DESCRIPTION
**Pull request by umsaleem**
This commits updates the Debian packaging bits to build scst-dbg package for kernel modules linked with debug kernel.

SCST kernel modules linked with debug kernel need to packaged differently so that they can co-exist with SCST package for modules built for production kernel.

The updates in packaging bits are minimal would only apply if KVERS environment variable is configured for debug kernel for TrueNAS SCALE.

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]
I have no objections here and I tested this and confirmed it worked. Even running it through our full CI, no obvious regressions were found. However, I'd like to wait for @bmeagherix final review before merge, please. (since he has taken over managing this particular repo)

With these changes, rebasing on upstream will produce a conflicts. I was trying to avoid that scenario with this repo but I understand that it's not always possible.

**Review by bmeagherix — APPROVED** [review]
FWIW, for a while a missing piece of the puzzle for me was the corresponding changes over in the _scale-build_ repo.

LGTM.

**Pull request merged**
